src/api/lib/tasks/dev/requests.rake

```diff
@@ -199,5 +199,357 @@ namespace :dev do
+
+            if repository['arch'].is_a?(Array)
+              repository['arch'].each do |architecture|
+                proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: architecture))
+              end
+            else
+              proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: repository['arch']))
+            end
+            proj_repo.save!
+            project.store(comment: "Added #{proj_repo.name} repository")
+          end
+
+          next
+        elsif repository['path']
+          project = Project.find_by_name(repository['path']['project'])
+          project = create(:project, name: repository['path']['project']) if project.nil?
+          proj_repo = project.repositories.find_or_initialize_by(name: repository['path']['repository'])
+        else
+          next
+        end
+        target_repo = repository['name']
+        target_repository = Repository.find_by_project_and_name(target_project.name, target_repo)
+        target_repository = create(:repository, project: target_project, name: target_repo) if target_repository.nil?
+        proj_repo.path_elements.find_or_initialize_by(link: target_repository)
+
+        if repository['arch'].is_a?(Array)
+          repository['arch'].each do |architecture|
+            proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: architecture))
+          end
+        else
+          proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: repository['arch']))
+        end
+        proj_repo.save!
+        project.store(comment: "Added #{proj_repo.name} repository")
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
+    def create_users_and_groups_from_meta(request_data:, users:)
+      print_message('creating users and groups from _meta ....')
+      # create users
+      if request_data['person'].present?
+        request_data['person'].each do |person|
+          create_user(login: person['userid'], users: users)
+        end
+      end
+
+      return if request_data['group'].nil?
+
+      if request_data['group'].is_a?(Array)
+        request_data['group'].each do |group|
+          next if Group.find_by(title: group['groupid'])
+
+          create(:group, title: group['groupid'])
+        end
+      else
+        group = Group.find_by(title: request_data['group']['groupid'])
+
+        create(:group, title: request_data['group']['groupid']) if group.nil?
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    def create_user(login:, users: [])
+      return if User.find_by(login: login).present?
+
+      users << create(:staff_user, login: login)
```

Comment:
Why staff users? And why assign this to this empty array?

Reply:
No specific reason. Just to make sure nothing breaks. Converted it to `confirmed_users`.
This empty users list was to keep track of the newly created users so they could be anonymized later on